pkg/set/set_manager.go

```diff
@@ -139,3 +147,32 @@ func (s *ManagedSet) genTags(additional []string) []string {
 
 	return append(additional, defaultTags...)
 }
+
+func (s *ManagedSet) emitUsageCounters(setDataList []countedSetData) {
+	for _, d := range setDataList {
+		var tags []string
+		if err := utils.ValidatePort(d.setData.Port); err != nil {
+			tags = []string{fmt.Sprintf("startip_endip:%v", d.setData.Port)}
+		} else if err := utils.ValidatePortRange(d.setData.PortRangeStart, d.setData.PortRangeEnd); err != nil {
+			tags = []string{fmt.Sprintf("startip_endip:%v-%v", d.setData.PortRangeStart, d.setData.PortRangeEnd)}
+		} else if err := utils.ValidatePrefix(d.setData.Prefix); err != nil {
+			tags = []string{fmt.Sprintf("startip_endip:%s", d.setData.Prefix)}
+		} else if err := utils.ValidateAddress(d.setData.Address); err != nil {
+			tags = []string{fmt.Sprintf("startip_endip:%s", d.setData.Address)}
+		} else if err := utils.ValidateAddressRange(d.setData.AddressRangeStart, d.setData.AddressRangeEnd); err != nil {
+			tags = []string{fmt.Sprintf("startip_endip:%s-%s", d.setData.AddressRangeStart, d.setData.AddressRangeEnd)}
+		} else {
+			s.logger.Warnf("invalid set data encountered while emitting counter metrics: %+v", d.setData)
+			continue
+		}
```

Comment:
I'm open to adding validation methods directly to SetData like `hasValidPortRange()`, `hasValidPrefix()` to perhaps make this a bit more readable

Reply:
To address the readability you can also do:

```go
switch {
case utils.ValidatePort(d.setData.Port) != nil:
  tags = []string{fmt.Sprintf("startip_endip:%v", d.setData.Port)}

case utils.ValidatePortRange(d.setData.PortRangeStart, d.setData.PortRangeEnd) != nil:
  tags = []string{fmt.Sprintf("startip_endip:%v-%v", d.setData.PortRangeStart, d.setData.PortRangeEnd)}
...

default:
  s.logger.Warnf("invalid set data encountered while emitting counter metrics: %+v", d.setData)
  continue
}
```

This approach removes a lot of the `if else` noise which is a pretty common patter for lots of `if else` branches with more then 2.

Reply:
thank you 🙏🏻